### ISSUE TITLE
[IO] [Kafka] a Kafka source submitted with invalid configuration stucks without consuming from Kafka topics

### ISSUE THREAD

**Issue by sijie** [apache/pulsar#3294]
#### Expected behavior

Create a Kafka source connector with invalid configuration should fail 

#### Actual behavior

Create a Kafka source connector with invalid configuration succeed.

`getstatus` show the connector is running.
the function log doesn't show any exceptions.

#### Steps to reproduce

You can use the same steps in #3293  to repro this issue. Change `localrun` to `create`.

#### System configuration
**Pulsar version**: 2.2.0

**Comment by ambition119** [comment]
Use 'source create' command we validate verify that the parameter key is included, but it does not verify that the value of the key is correct, so create sucessflully. If value is error, should we return the create source failure?
kafka.yaml info:
```file
configs:
    bootstrapServers: "xxxxxxx"
    groupId: "pulsarUTset"
    topic: "test_kafka_topic"
    fetchMinBytes: 1
    autoCommitEnabled: "true"
    autoCommitIntervalMs: 1000
    sessionTimeoutMs: 3000
```
command like :
```shell
./bin/pulsar-admin source create --source-config-file ~/code/pulsar/pulsar_ms/file/kafka.yaml -t kafka --destination-topic-name dest_topic --name kafka_fn
"Created successfully"
```
when I getstatus return info: 
```shell
$ ./bin/pulsar-admin source getstatus  --name kafka_fn
{
  "numInstances" : 1,
  "numRunning" : 1,
  "instances" : [ {
    "instanceId" : 0,
    "status" : {
      "running" : true,
      "error" : "",
      "numRestarts" : 0,
      "numReceivedFromSource" : 0,
      "numSystemExceptions" : 0,
      "latestSystemExceptions" : [ ],
      "numSourceExceptions" : 0,
      "latestSourceExceptions" : [ ],
      "numWritten" : 0,
      "lastReceivedTime" : 0,
      "workerId" : "c-standalone-fw-localhost-8080"
    }
  } ]
}
```

**Comment by sijie** [comment]
@ambition119 I think my main concern here is the create operation succeed and get status shows the function instance is running. what I expect to see is get status should show the function instance is not running.

**Comment by liangyuanpeng** [comment]
It may already be working in version 2.3.0
i tried and i got it
```

  "numInstances" : 1,
  "numRunning" : 0,
  "instances" : [ {
    "instanceId" : 0,
    "status" : {
      "running" : false,
      "error" : "UNAVAILABLE: io exception",
      "numRestarts" : 0,
      "numReceivedFromSource" : 0,
      "numSystemExceptions" : 0,
      "latestSystemExceptions" : [ ],
      "numSourceExceptions" : 0,
      "latestSourceExceptions" : [ ],
      "numWritten" : 0,
      "lastReceivedTime" : 0,
      "workerId" : "c-standalone-fw-lan-Inspiron-5423-8080"
    }
  } ]
}
```

**Comment by sijie** [comment]
@liangyuanpeng yes there were bunch of fixes on kafka source in 2.3.0

**Comment by sijie** [comment]
The issue has been fixed in 2.3.0
